backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/PostgresLegalPeriodicalEditionRepositoryImpl.java

```diff
@@ -51,26 +51,33 @@ public LegalPeriodicalEdition save(LegalPeriodicalEdition legalPeriodicalEdition
     return LegalPeriodicalEditionTransformer.transformToDomain(repository.save(edition));
   }
 
-  private void deleteDocUnitLinksForDeletedReferences(
-      LegalPeriodicalEdition legalPeriodicalEdition) {
-    var oldEdition = repository.findById(legalPeriodicalEdition.id());
-    if (oldEdition.isPresent()) {
-      // Ensure it's removed from DocumentationUnit's references
-      for (ReferenceDTO reference : oldEdition.get().getReferences()) {
-        // skip all existing references
-        if (legalPeriodicalEdition.references().stream()
-            .anyMatch(newReference -> newReference.id().equals(reference.getId()))) {
-          continue;
-        }
-        // delete all deleted references
-        documentationUnitRepository
-            .findById(reference.getDocumentationUnit().getId())
-            .ifPresent(
-                docUnit -> {
-                  docUnit.getReferences().remove(reference);
-                  documentationUnitRepository.save(docUnit);
-                });
+  private void deleteDocUnitLinksForDeletedReferences(LegalPeriodicalEdition updatedEdition) {
```

Comment:
Alternatively, we could add a `deleteReference` endpoint in the controller specifically for `Reference` that only takes an `id`. This way, we could have avoided the filtering out and start directly by the desired item, not having to send the entire list of references on deletion.